### PR TITLE
hepmc3: add version 3.2.4

### PR DESCRIPTION
**Pull request by vvolkl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hepmc3/package.py [hunk 1]
```diff
@@ -42,8 +43,6 @@ class Hepmc3(CMakePackage):
     def cmake_args(self):
         spec = self.spec
         args = [
-            '-Dmomentum:STRING=GEV',
-            '-Dlength:STRING=MM',
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/hepmc3/package.py:46]: I assume these are needed to install the older versions? You should move the definitions to an `if spec.satisfies('@:3.2.3'):`

File: var/spack/repos/builtin/packages/hepmc3/package.py [hunk 2]
```diff
@@ -61,4 +60,5 @@ def cmake_args(self):
 
         if self.spec.satisfies('+rootio'):
             args.append('-DROOT_DIR={0}'.format(self.spec['root'].prefix))
+        args.append("-DHEPMC3_ENABLE_TEST={0}".format(self.run_tests))
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/hepmc3/package.py:63]: This is a good opportunity to transition to the cmake helper functions (which add the `BOOL` type for boolean python values and join lists with `;`):
```suggestion
        args.append(self.define('HEPMC3_ENABLE_TEST', self.run_tests))
```

**Review by sethrj — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/hepmc3/package.py (hunk 1 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/hepmc3/package.py:46]: No, those apply to hepmc 2, which has its own recipe.

File: var/spack/repos/builtin/packages/hepmc3/package.py (hunk 2 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/hepmc3/package.py:63]: Sure, I'll update those.

**Review by sethrj — APPROVED** [review]

File: var/spack/repos/builtin/packages/hepmc3/package.py (hunk 2 above)
Comment (sethrj) [reply var/spack/repos/builtin/packages/hepmc3/package.py:63]: You can just worry about it for next time if the opportunity arises -- this one's pipeline has passed so I'll just merge it. Thanks!

**Pull request merged**